lib/collection/src/wal_delta.rs

```diff
@@ -192,13 +204,16 @@ fn resolve_wal_delta(
     // Scroll back over the WAL and find a record that covered all clocks
     // Drain satisfied clocks from the recovery point until we have nothing left
     log::trace!("Resolving WAL delta for: {recovery_point}");
-    let delta_from = local_wal
-        .lock()
-        .read_from_last(true)
+
+    let wal = local_wal.lock();
+
+    let delta_from = wal
+        .read(wal.first_closed_index())
+        .rev()
```

Comment:
This is where I'd prefer the following:

```suggestion
        .read_all(true)
        .rev()
```

Reply:
Hmmm. Maybe you're right. Ok, I'll return the `read_all`. 👍